server-side/tools/deploy-changes-since/Deploy-ChangesSince.ps1

```diff
@@ -0,0 +1,23 @@
+# Deploy-ChangesSince.ps1
+# 
+# Example usage: ./Deploy-ChangesSince.ps1 -EarliestChangeDate "2019-06-20" -DestServer "ultra-gro.shift3sandbox.com" -DestBasePath "Ultra-Gro"
+# Note that you should run this from the directory on your local machine where
+# your source code is located. Also, make sure that you have the latest source
+# (preferably from the master branch) checked out before you run this script!
+# 
+# Finally, make sure to give the correct path to Deploy-ChangesSince.ps1,
+# either absolute or relative. Even if you're in the same directory as this
+# file, PowerShell won't run it unless you specify ./ ahead of the filename.
+
+param([DateTime]$EarliestChangeDate, [String]$DestServer, [String]$DestBasePath)
+
+[String]$SrcBasePath = Get-Location
+Get-ChildItem -Recurse | Where-Object {!$_.PSIsContainer -and ($_.LastWriteTime -gt $EarliestChangeDate)} | Sort-Object LastWriteTime -Descending | ForEach-Object {
+    [String]$SrcPath = $_.FullName
+    [String]$SubPath = $SrcPath.Replace($SrcBasePath, "")
+    [String]$DestPath = $DestServer + ":" + $DestBasePath + $SubPath
+    [String]$CmdToRun = "scp $SrcPath $DestPath"
```

Comment:
I appreciate the effort put into this script, but i think there are a number of problems we would want to solve. Typically deploying a django project involves.

1. Check out the latest code on the server, typically we use a private key for deploy. `git` checkout ensures that if we delete a file or rename a file that those changes are tracked, `scp` does not track this change. `scp` also does not take into account editor files that are gitignored, or other files like `.pyc` files that are also typically gitignored as well. A `pyc` compiles file on mac osx is probably not going to be compatible with an ubuntu system. 
2. Prepare virtualenv, install new dependencies from `requirements.txt` or `Pipfile`
3. Run `./manage.py collectstatic` after checkout.
4. Run any new migrations.
5. Run a rolling reload across the services involved, typically in the past we would use supervisord as the process manager, with systemd however we may use a systemd service file instead.

`fabric` is usually the defacto standard in the python world for automating these steps.

Reply:
Used to use fabric to deploy django, but now I've switch over to docker. Manual deploy would be as follows:

1. Build docker image locally, export image into a tarball
2. scp docker image to server
3. ssh into server, import docker image
4. kick docker service, probably with docker compose

Migrations and other management tasks can be ran from the server or locally with server environment variables.

Reply:
Ah right, maybe we can get the dockerfile in here @stephengtuggy ?

Reply:
@DropsOfSerenity docker files now included (four of them). Let me know what you think.

Reply:
🙌 For django folks!